middleware/fastcgi/fastcgi.go

```diff
@@ -292,6 +289,20 @@ type Rule struct {
 	EnvVars [][2]string
 }
 
+// canSplit checks if path can split into two based on rule.SplitPath.
+func (r Rule) canSplit(path string) bool {
+	return r.splitPos(path) >= 0
+}
+
+// splitPos returns the index where path should be split
+// based on rule.SplitPath.
+func (r Rule) splitPos(path string) int {
+	if middleware.CaseSensitivePath {
+		return strings.Index(path, r.SplitPath)
+	}
+	return strings.Index(strings.ToLower(path), r.SplitPath)
```

Comment:
I take that as a poor assumption from me. I was relying on the users to use lowercase in the Caddyfile as most people do.